infinite loop with fortran parser on C file

How to reproduce
```
./ctags.20151012 --language-force=fortran ctags/Units/parser-c.r/enum.c.d/input.c
```

It's found by forcing language as fortran against universal-ctags unit test files.

BTW You can do the same with fuzz target:
```
b% make fuzz LANGUAGES=Fortran
make fuzz LANGUAGES=Fortran
Semi-fuzzing (Fortran)                                      
............................................
[timeout Fortran]                       ./Units/cxx11enum.cpp.d/input.cpp

Makefile:399: recipe for target 'fuzz' failed
make: *** [fuzz] Error 1
```

(I'm looking for better name for the target. It is not "fuzz".)

SRHINK=1 spots the token.

```
b% make fuzz LANGUAGES=Fortran SHRINK=1
make fuzz LANGUAGES=Fortran SHRINK=1
Semi-fuzzing (Fortran)                                      
............................................
[timeout Fortran]                       ./Units/cxx11enum.cpp.d/input.cpp
Shrinking ./Units/cxx11enum.cpp.d/input.cpp as Fortran
............................................................
[0        351]...timeout(124)
[0          0]...ok(0)
[0        351]...timeout(124)
[0        176]...timeout(124)
[0         89]...timeout(124)
[0         46]...timeout(124)
[0         25]...timeout(124)
[0         15]...timeout(124)
[0         10]...timeout(124)
[0          8]...timeout(124)
[0          7]...timeout(124)
[0          6]...timeout(124)
[0          5]...timeout(124)
[0          4]...ok(0)
[0          5]...timeout(124)
[2          5]...ok(0)
[1          5]...timeout(124)
Minimal badinput: ./Units/cxx11enum.cpp.d/SHRINK-Fortran.tmp
............................................................
enum

Makefile:399: recipe for target 'fuzz' failed
make: *** [fuzz] Error 1
```

--- a/Units/parser-fortran.r/bug629.d/args.ctags
+++ b/Units/parser-fortran.r/bug629.d/args.ctags
@@ -1,0 +1,1 @@
+--language-force=Fortran

--- a/Units/parser-fortran.r/bug629.d/expected.tags
+++ b/Units/parser-fortran.r/bug629.d/expected.tags
@@ -1,0 +1,1 @@
+test_enumeration	input.c	/^enum test_enumeration /;"	E

--- a/Units/parser-fortran.r/bug629.d/input.c
+++ b/Units/parser-fortran.r/bug629.d/input.c
@@ -1,0 +1,5 @@
+enum test_enumeration {
+    ENUM_1 = 1,
+    ENUM_2,
+    ENUM_3
+};

--- a/parsers/fortran.c
+++ b/parsers/fortran.c
@@ -2052,7 +2052,8 @@ static void parseEnumBlock (tokenInfo *const token)
 		makeFortranTag (name, TAG_ENUM);
 	skipToNextStatement (token);
 	ancestorPush (name);
-	while (! isKeyword (token, KEYWORD_end))
+	while (! isKeyword (token, KEYWORD_end) &&
+		   ! isType(token, TOKEN_EOF))
 	{
 		if (isTypeSpec (token))
 			parseTypeDeclarationStmt (token);

